test/Shared/UserManagerTestBase.cs

```diff
@@ -82,15 +83,16 @@ protected RoleManager<TRole> CreateRoleManager(object context = null, IServiceCo
         protected abstract void AddUserStore(IServiceCollection services, object context = null);
         protected abstract void AddRoleStore(IServiceCollection services, object context = null);
 
-        protected TUser CreateTestUser(string namePrefix = "")
-        {
-            return new TUser() { UserName = namePrefix + Guid.NewGuid().ToString() };
-        }
+        protected abstract void SetUserPasswordHash(TUser user, string hashedPassword);
 
-        protected TRole CreateRole(string namePrefix = "")
-        {
-            return new TRole() { Name = namePrefix + Guid.NewGuid().ToString() };
-        }
+        protected abstract TUser CreateTestUser(string namePrefix = "", string email = "", string phoneNumber = "",
+            bool lockoutEnabled = false, DateTimeOffset? lockoutEnd = null, bool useNamePrefixAsUserName = false);
+
+        protected abstract TRole CreateTestRole(string roleName = "");
+
+        protected abstract Expression<Func<TUser, bool>> UserNamePredicate(string userName, bool contains = false);
```

Comment:
1. I would rather simplify the tests to just succeed with simple equality and get rid of the contains flag on both predicates.
2. Contains is not the same as StartsWith.